### PR TITLE
doc pitfalls: add a note about limits definition

### PR DESCRIPTION
**Pull request by hervenicol**
Towards https://github.com/giantswarm/giantswarm/issues/25998

Try to explain why some containers use more resources than expected.
A bit tricky to explain... waiting for your feedback ;)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,11 +54,22 @@ It will automatically adjust the pod's memory `requests` between `100Mi` and `50
...
 * VPA increases it to `requests: 400Mi`, it also sets `limits: 600Mi`.
 * Notice that now the RAM limit is higher than the VPA's `maxAllowed`.
 
+### Containers grow bigger than available resources
+
+There's 2 things that you may aim for with VPA:
+- make sure your container has enough resources to work (`requests` management)
+- make sure your container fits on your nodes (`limits` management)
+
+When a container uses too much resources, keep in mind that [VPA sets the container's requests](#vpa-sets-the-containers-requests) and that `limits` are set proportionally.
```
Comment (QuentinBisson) [inline README.md:69]: WDYT about 

```suggestion
When a container uses too much resources, keep in mind that [VPA sets the container's requests](#vpa-sets-the-containers-requests) and that `limits` are set proportionally following on the original request/limit ratio
```

File: README.md [hunk 2]
```diff
@@ -54,11 +54,22 @@ It will automatically adjust the pod's memory `requests` between `100Mi` and `50
...
+
+If you want to make sure your container won't use more resources than `maxAllowed`, set `limits`=`requests` at container creation.
```
Comment (QuentinBisson) [inline README.md:71]: I would add something along the lines of:

When setting the maxAllowed (max allowed requests) values, beware that the requests or vpa computed limits do not go over the node available resources or that the lim

Comment (hervenicol) [reply README.md:75]: oooh, yes, I was totally missing the point of "my pod isn't schedulable anymore" :exploding_head: 
I added something, hope it covers it.

Comment (QuentinBisson) [reply README.md:75]: I love it :)

**Review by QuentinBisson — APPROVED** [review]

**Pull request merged**